src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/MediaTests.ps1

```diff
@@ -0,0 +1,194 @@
+  $storageAccount2 = CreateStorageAccount $rgname $storageAccountName2 $location
+
+  <# Check name availability #>
+  $accountName = "med" + $rgname
+  $availability = Get-AzureRmMediaServiceNameAvailability -AccountName $accountName
+  Assert-AreEqual $true $availability.nameAvailable
+
+  <# Create a media service with specifiying the primary storage account only #>
+  $accountName = "med" + $rgname
+  $tags = @{"tag1" = "value1"; "tag2" = "value2"}
+  $storageAccount1 = GetStorageAccount -ResourceGroupName $rgname -Name $storageAccountName1
+  $mediaService = New-AzureRmMediaService -ResourceGroupName $rgname -AccountName $accountName -Location $location -StorageAccountId $storageAccount1.Id -Tags $tags
+  Assert-NotNull $mediaService
+  Assert-AreEqual $accountName $mediaService.AccountName
+  Assert-AreEqual $rgname $mediaService.ResourceGroupName
+  Assert-AreEqual $location $mediaService.Location
+  Assert-Tags $tags $mediaService.Tags
+  Assert-AreEqual $storageAccountName1 $mediaService.StorageAccounts[0].AccountName
+  Assert-AreEqual $true $mediaService.StorageAccounts[0].IsPrimary
+  Assert-AreEqual $rgname $mediaService.StorageAccounts[0].ResourceGroupName
+
+  $availability = Get-AzureRmMediaServiceNameAvailability -AccountName $accountName
+  Assert-AreEqual $false $availability.nameAvailable
+
+  <# Get a media service with specifying resource group only #>
+  $mediaServices = Get-AzureRmMediaService -ResourceGroupName $rgname 
+  Assert-NotNull $mediaServices
+  Assert-AreEqual 1 $mediaServices.Count
+  Assert-AreEqual $accountName $mediaServices[0].AccountName
+  Assert-AreEqual $rgname $mediaServices[0].ResourceGroupName
+  Assert-AreEqual $location $mediaServices[0].Location
+  Assert-AreEqual $storageAccountName1 $mediaServices[0].StorageAccounts[0].AccountName
+  Assert-AreEqual $true $mediaService.StorageAccounts[0].IsPrimary
+  Assert-AreEqual $rgname $mediaServices[0].StorageAccounts[0].ResourceGroupName
+
+  <# Get a media service with specifying resource group and account name #>
+  $mediaService = Get-AzureRmMediaService -ResourceGroupName $rgname -AccountName $accountName
+  Assert-NotNull $mediaService
+  Assert-AreEqual $accountName $mediaService.AccountName
+  Assert-AreEqual $rgname $mediaService.ResourceGroupName
+  Assert-AreEqual $location $mediaService.Location
+  Assert-AreEqual $storageAccountName1 $mediaService.StorageAccounts[0].AccountName
+  Assert-AreEqual $true $mediaService.StorageAccounts[0].IsPrimary
+  Assert-AreEqual $rgname $mediaService.StorageAccounts[0].ResourceGroupName
+
+  <# Set a media service #>
+  $tagsUpdated = @{"tag3" = "value3"; "tag4" = "value4"}
+  $storageAccount2 = GetStorageAccount -ResourceGroupName $rgname -Name $storageAccountName2
+  $primaryStorageAccount = New-Object PSObject -Property @{"Id" = $storageAccount1.Id; "IsPrimary" = $true; }
+  $secondaryStorageAccount = New-Object PSObject -Property @{"Id" = $storageAccount2.Id; "IsPrimary" = $false; }
```

Comment:
>New-Object PSObject -Property @{"Id" = $storageAccount2.Id; "IsPrimary" = $false; } [](start = 29, length = 83)

not a good way for end users. You should expose a cmdlet that will create this object.